server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -242,7 +242,7 @@ public Object2IntMap<String> getDatasourceToUnavailableSegmentCount()
     final Iterable<DataSegment> dataSegments = metadataManager.segments().iterateAllUsedSegments();
     for (DataSegment segment : dataSegments) {
       SegmentReplicaCount replicaCount = segmentReplicationStatus.getReplicaCountsInCluster(segment.getId());
-      if (replicaCount != null && replicaCount.totalLoaded() > 0) {
+      if (replicaCount != null && (replicaCount.totalLoaded() > 0 || replicaCount.required() == 0)) {
```

Comment:
Should we also document this change here : 
`/docs/operations/metrics.md#L333`. Thinking more about it on how the doc would look, should we add a new metric type if replicaCount.required==0 ?
The reason I am saying this is that users on the regular sql/native endpoint will never see the data from used segment in deep storage ie`required==0` . They might get thrown off because of this. 

I feel the current doc change for the metric fails to capture this nuance. 
If we try to explain this nuance, it becomes complicated.
A new metric `segment/deepStorage/count` would probably make more sense 
no ?
Wdyt ?

Reply:
Yeah, I did have similar thoughts.

Following the approach used in this PR, no segment would ever truly be considered unavailable since it can always be queried using MSQ-query-from-deep-storage.

The source of the confusion is the definition of `unavailable/count` itself.
Does `unavailable` denote a segment
1. that should be loaded but is not?
OR
2. that should be queryable but is not?

The metric `underReplicated/count` corresponds with option 1 and this PR makes `unavailable/count` use option 1 too.

The docs currently include both the flavors in the definition of this metric (which is correct for the native SQL engine):
> Number of unique segments __left to load__ until all used segments are __available for queries__.

@georgew5656 , to eliminate the confusion, as @cryptoe suggests, I think we should:
- keep this metric unchanged (in case there are downstream apps relying on its values being reported correctly for the native engine)
- add a new metric for MSQ cases
- update the new info in the docs

Reply:
i don't think this is changing the existing behavior afaik. before this change, if a segment was not loaded via a load rule it would not be included in segment/unavailable/count, now the behavior would be the same.

the difference would be that a segment that is not loaded onto a historical (hot rule) but is covered by a cold rule is no longer counted in segment/unavailable/count.